### DISCUSSION TITLE
Change log in a second comment

### DISCUSSION THREAD

**Discussion by aschepler** [davidquinn/psi-header#16] [Ideas]
Some existing files in our project at work start with two comments, separating the copyright legalese and the change log. And the change log is earliest-to-latest:

```cpp
/*
 Copyright (C) 1995-2020 The Company
 Ipsum lorem blah blah blah
*/

/*
LIBRARY: stufflib
FILE: CommonStuff.hpp
CHANGE LOG:
Name      Date         Comment
========  ===========  =====================================
doeja     02-May-1995  Initial version.
blackwc   10-Jan-2000  Improved const correctness.
jinglejj  20-Apr-2020  Added convert_stuff overload.
*/
```

If the `changeLogNaturalOrder` setting is true, the Ctrl-Alt-C Ctrl-Alt-C command will insert a change comment at the end of the first header with copyright. If the `changeLogNaturalOrder` setting is false, I think I could have it insert in the correct block, but at the start of the history instead of the end.

I'm not sure how common this style is. Would it be a reasonable change to support a configured mode where Ctrl-Alt-C Ctrl-Alt-C will look for a comment containing `changeLogCaption` but then add to the end of that comment (minus `changeLogFooterLineCount`)? Maybe just do this if both `changeLogCaption` and `changeLogNaturalOrder=true` are specified; or maybe add some new config setting.

**Comment by davidquinn** [comment]
Hi,

I am sorry but I did not see this post before now. I appreciate that you might have moved on by now due to the lack of activity here, but here goes...

I don't think this is a common setup and I have never had a similar request before,  but I can see if it is possible to deal with and if it makes sense to add something that recognises this.

Let me know if this is still an issue and if you want me to take a closer look.

Regards
David
